x-pack/plugins/apm/public/components/app/transaction_details/Distribution/index.tsx

```diff
@@ -133,15 +134,22 @@ export function TransactionDistribution({
   const xMax = d3.max(buckets, (d) => d.x0) || 0;
   const timeFormatter = getDurationFormatter(xMax);
 
-  const tooltipProps: SettingsSpec['tooltip'] = {
-    headerFormatter: (tooltip: TooltipValue) => {
-      const serie = buckets.find((bucket) => bucket.x0 === tooltip.value);
-      if (serie) {
-        const xFormatted = timeFormatter(serie.x);
-        const x0Formatted = timeFormatter(serie.x0);
-        return `${x0Formatted.value} - ${xFormatted.value} ${xFormatted.unit}`;
-      }
-      return `${timeFormatter(tooltip.value)}`;
+  const distributionMap = keyBy(distribution?.buckets, 'key');
+  const bucketsMap = keyBy(buckets, 'x0');
+
+  const tooltip: SettingsSpec['tooltip'] = {
+    customTooltip: (props: TooltipInfo) => {
+      const datum = props.header?.datum as IChartPoint;
+      const selectedDistribution = distributionMap[datum?.x0];
+      const serie = bucketsMap[datum?.x0];
+      return (
+        <CustomTooltip
+          {...props}
+          hasSamples={isEmpty(selectedDistribution?.samples)}
```

Comment:
Here when `selectedDistribution` has no `samples`, then `hasSamples` becomes `true`...seems opposite of the prop name.